cirq/linalg/predicates.py

```diff
@@ -189,3 +190,40 @@ def allclose_up_to_global_phase(
 
     # Should now be equivalent.
     return np.allclose(a=a, b=b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+
+def binary_sub_tensor_slice(index_bits: int,
```

Comment:
this name is a bit awkward (it's not clear what the sub modifies to me).  restricted_tensor_slice?  I might also make it more general with a changable base (besides 2), but not really needed yet.

Reply:
Renamed to `slice_for_qubits_equal_to`.

Reply:
Perfect!